### ISSUE TITLE
auto_pause_notifications_parameters - not apply

### ISSUE THREAD

**Issue by elihai-risk** [PagerDuty/terraform-provider-pagerduty#574]
Hi there,

This Feature doesn't apply by Terraform
in the Terraform plan it show it is going to add it but in the UI it's still off.
if i changing manually it doesn't show me drift

Terraform v1.2.3

PagerDuty providers - 2.6.2

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### Debug Output
Please provider a link to a GitHub Gist containing the complete debug output: https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

### Expected Behavior
What should have happened?

### Actual Behavior
What actually happened?

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? Custom version of OpenStack? Tight ACLs?

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- GH-1234

**Comment by pdecat** [comment]
Hi @elihai-risk,

could you please also share an extract of your `.tf` terraform configuration file where you are using this feature?

It should be something like this:

```
resource "pagerduty_service" "foo" {
	name                    = "foo"
	description             = "foo"
	auto_resolve_timeout    = 1800
	acknowledgement_timeout = 1800
	escalation_policy       = pagerduty_escalation_policy.foo.id
	alert_creation          = "create_alerts_and_incidents"
	auto_pause_notifications_parameters {
		enabled = true
		timeout = 300
    }
}
```

Also, debug logs produced with the `TF_LOG=debug` environment variable set (**redact any confidential information from that log before sharing**) are welcome:

> Please provider a link to a GitHub Gist containing the complete debug output: https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

**Comment by elihai-risk** [comment]
`resource "pagerduty_service" "service" {
  for_each                = { for technical_service in local.technical_services : technical_service.name => technical_service }
  name                    = "${each.value.name}-${var.env}"
  escalation_policy       = lookup(each.value, "escalation_policy", null) == null ? data.pagerduty_escalation_policy.escalation_policy.id : data.pagerduty_escalation_policy.service_escalation_policy[each.value.name].id
  alert_creation          = "create_alerts_and_incidents"
  acknowledgement_timeout = lookup(each.value, "acknowledgement_timeout", "null")
  auto_resolve_timeout    = "null"

  dynamic "alert_grouping_parameters" {
    for_each = var.alert_grouping == true ? [1] : []
    content {
      type = "intelligent"
      config {
        timeout = null
        fields  = null
      }
    }
  }

  dynamic "incident_urgency_rule" {
    for_each = var.env != "staging" ? [1] : []
    content {
      type    = "constant"
      urgency = "severity_based"
    }
  }

  dynamic "incident_urgency_rule" {
    for_each = var.env == "staging" ? [1] : []
    content {
      type = "use_support_hours"

      during_support_hours {
        type    = "constant"
        urgency = "severity_based"
      }

      outside_support_hours {
        type    = "constant"
        urgency = "low"
      }
    }
  }

  dynamic "support_hours" {
    for_each = var.env == "staging" ? [1] : []
    content {
      type         = "fixed_time_per_day"
      time_zone    = "Asia/Jerusalem"
      start_time   = "10:00:00"
      end_time     = "19:00:00"
      days_of_week = [7, 1, 2, 3, 4]
    }
  }

  auto_pause_notifications_parameters {
    enabled = true
    timeout = 300
  }
}`

**Comment by elihai-risk** [comment]
module.technical_service_staging.pagerduty_service.service["account-event-linking-service"]: Modifying... [id=PVJKT7N]
2022-09-29T13:42:46.012+0300 [INFO]  Starting apply for module.technical_service_staging.pagerduty_service.service["account-event-linking-service"]
2022-09-29T13:42:46.013+0300 [DEBUG] module.technical_service_staging.pagerduty_service.service["account-event-linking-service"]: applying the planned Update change
2022-09-29T13:42:46.013+0300 [DEBUG] module.technical_service_staging.pagerduty_service.service["customer-view-ui"]: applying the planned Update change
2022-09-29T13:42:46.014+0300 [DEBUG] module.technical_service_staging.pagerduty_service.service["ipv-classification-consumer"]: applying the planned Update change
2022-09-29T13:42:46.018+0300 [INFO]  provider.terraform-provider-pagerduty_v2.6.2: 2022/09/29 13:42:46 ===== PagerDuty Cache Skipping Init =====: timestamp=2022-09-29T13:42:46.018+0300
2022-09-29T13:42:46.020+0300 [INFO]  provider.terraform-provider-pagerduty_v2.6.2: 2022/09/29 13:42:46 [DEBUG] PagerDuty - Preparing GET request to /abilities with body: %!s(<nil>): timestamp=2022-09-29T13:42:46.020+0300
2022-09-29T13:42:46.025+0300 [INFO]  provider.terraform-provider-pagerduty_v2.6.2: 2022/09/29 13:42:46 [DEBUG] PagerDuty API Request Details:
---[ REQUEST ]---------------------------------------
GET /abilities HTTP/1.1
Host: api.pagerduty.com
User-Agent: heimweh/go-pagerduty(terraform)
Accept: application/vnd.pagerduty+json;version=2
Authorization: Token token=xxxxxxxxxx
Content-Type: application/json
Accept-Encoding: gzip


-----------------------------------------------------: timestamp=2022-09-29T13:42:46.025+0300
2022-09-29T13:42:47.270+0300 [INFO]  provider.terraform-provider-pagerduty_v2.6.2: 2022/09/29 13:42:47 [DEBUG] PagerDuty API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Access-Control-Allow-Headers: Authorization, Content-Type, AuthorizationOauth, X-EARLY-ACCESS
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers:
Access-Control-Max-Age: 1728000
Cache-Control: max-age=0, private, must-revalidate
Content-Type: application/json
Date: Thu, 29 Sep 2022 10:42:47 GMT
Etag: W/"38210395dd4b3bb2ecca0acd03c9cb3a"
Feature-Policy: accelerometer 'none'; camera 'none'; geolocation 'none'; gyroscope 'none'; magnetometer 'none'; microphone 'none'; payment 'none'; usb 'none'
Referrer-Policy: strict-origin-when-cross-origin
Referrer-Policy: strict-origin-when-cross-origin
Server: nginx
Strict-Transport-Security: max-age=31536000; includeSubDomains
Vary: Accept-Encoding
X-Content-Type-Options: nosniff
X-Request-Id: 97b63e651db7aee5690be62cdd256497
X-Xss-Protection: 1; mode=block

{
 "abilities": [
  "sso",
  "advanced_reports",
  "teams",
  "read_only_users",
  "team_responders",
  "service_support_hours",
  "urgencies",
  "manage_schedules",
  "manage_api_keys",
  "coordinated_responding",
  "response_automation",
  "using_alerts_on_any_service",
  "event_rules",
  "global_rulesets",
  "dynamic_notifications",
  "beta_custom_actions",
  "preview_incident_alert_split",
  "permissions_service",
  "advanced_permissions",
  "preview_intelligent_alert_grouping",
  "time_based_alert_grouping",
  "mobile_schedules",
  "responseplays_mobile_ios_v1",
  "responseplays_mobile_android_v1",
  "liveness_ios_v1",
  "liveness_android_v1",
  "preview_incident_priority",
  "incident_priority_activated",
  "business_visibility",
  "beta_conference_calls",
  "similar_incidents",
  "tags",
  "subscribers_status_updates",
  "status_dashboard_advanced",
  "advanced_ticketing_integrations",
  "incident_correlation",
  "status_update_integration",
  "manual_escalation",
  "change_events",
  "event_intelligence_datadog_widget",
  "mobile_incident_redesign",
  "mobile_home_screen",
  "mobile_maintenance_windows",
  "features_in_use_preventing_downgrade_to",
  "products_requiring_cleanup"
 ]
}
-----------------------------------------------------: timestamp=2022-09-29T13:42:47.270+0300
2022-09-29T13:42:47.270+0300 [INFO]  provider.terraform-provider-pagerduty_v2.6.2: 2022/09/29 13:42:47 [INFO] PagerDuty client configured: timestamp=2022-09-29T13:42:47.270+0300
2022-09-29T13:42:47.272+0300 [INFO]  provider.terraform-provider-pagerduty_v2.6.2: 2022/09/29 13:42:47 [INFO] Updating PagerDuty service PAJ8G0E: timestamp=2022-09-29T13:42:47.272+0300
2022-09-29T13:42:47.272+0300 [INFO]  provider.terraform-provider-pagerduty_v2.6.2: 2022/09/29 13:42:47 [INFO] Updating PagerDuty service PVJKT7N: timestamp=2022-09-29T13:42:47.272+0300
2022-09-29T13:42:47.272+0300 [INFO]  provider.terraform-provider-pagerduty_v2.6.2: 2022/09/29 13:42:47 [INFO] Updating PagerDuty service PKR5W0D: timestamp=2022-09-29T13:42:47.272+0300
2022-09-29T13:42:47.272+0300 [INFO]  provider.terraform-provider-pagerduty_v2.6.2: 2022/09/29 13:42:47 [INFO] Updating PagerDuty service PSNZSUK: timestamp=2022-09-29T13:42:47.272+0300
2022-09-29T13:42:47.272+0300 [INFO]  provider.terraform-provider-pagerduty_v2.6.2: 2022/09/29 13:42:47 [DEBUG] PagerDuty - Preparing PUT request to /services/PSNZSUK with body: {"service":{"acknowledgement_timeout":null,"alert_creation":"create_alerts_and_incidents","alert_grouping":"intelligent","alert_grouping_parameters":{"type":"intelligent","config":{"timeout":0,"aggregate":""}},"auto_pause_notifications_parameters":{"enabled":true,"timeout":180},"auto_resolve_timeout":null,"description":"Managed by Terraform","escalation_policy":{"id":"P6D1FFL","type":"escalation_policy_reference"},"incident_urgency_rule":{"during_support_hours":{"type":"constant","urgency":"severity_based"},"outside_support_hours":{"type":"constant","urgency":"low"},"type":"use_support_hours"},"name":"customer-view-ui-staging","scheduled_actions":[null],"support_hours":{"days_of_week":[7,1,2,3,4],"end_time":"19:00:00","start_time":"10:00:00","time_zone":"Asia/Jerusalem","type":"fixed_time_per_day"}}}: timestamp=2022-09-29T13:42:47.272+0300
2022-09-29T13:42:47.272+0300 [INFO]  provider.terraform-provider-pagerduty_v2.6.2: 2022/09/29 13:42:47 [INFO] Updating PagerDuty service PA7DZLC: timestamp=2022-09-29T13:42:47.272+0300
2022-09-29T13:42:47.272+0300 [INFO]  provider.terraform-provider-pagerduty_v2.6.2: 2022/09/29 13:42:47 [DEBUG] PagerDuty - Preparing PUT request to /services/PA7DZLC with body: {"service":{"acknowledgement_timeout":null,"alert_creation":"create_alerts_and_incidents","alert_grouping":"intelligent","alert_grouping_parameters":{"type":"intelligent","config":{"timeout":0,"aggregate":""}},"auto_pause_notifications_parameters":{"enabled":true,"timeout":180},"auto_resolve_timeout":null,"description":"Managed by Terraform","escalation_policy":{"id":"P6D1FFL","type":"escalation_policy_reference"},"incident_urgency_rule":{"during_support_hours":{"type":"constant","urgency":"severity_based"},"outside_support_hours":{"type":"constant","urgency":"low"},"type":"use_support_hours"},"name":"account-secure-webapp-ui-staging","scheduled_actions":[null],"support_hours":{"days_of_week":[7,1,2,3,4],"end_time":"19:00:00","start_time":"10:00:00","time_zone":"Asia/Jerusalem","type":"fixed_time_per_day"}}}: timestamp=2022-09-29T13:42:47.272+0300
2022-09-29T13:42:47.272+0300 [INFO]  provider.terraform-provider-pagerduty_v2.6.2: 2022/09/29 13:42:47 [INFO] Updating PagerDuty service PZW8TJA: timestamp=2022-09-29T13:42:47.272+0300
2022-09-29T13:42:47.272+0300 [INFO]  provider.terraform-provider-pagerduty_v2.6.2: 2022/09/29 13:42:47 [DEBUG] PagerDuty - Preparing PUT request to /services/PZW8TJA with body: {"service":{"acknowledgement_timeout":null,"alert_creation":"create_alerts_and_incidents","alert_grouping":"intelligent","alert_grouping_parameters":{"type":"intelligent","config":{"timeout":0,"aggregate":""}},"auto_pause_notifications_parameters":{"enabled":true,"timeout":180},"auto_resolve_timeout":null,"description":"Managed by Terraform","escalation_policy":{"id":"P6D1FFL","type":"escalation_policy_reference"},"incident_urgency_rule":{"during_support_hours":{"type":"constant","urgency":"severity_based"},"outside_support_hours":{"type":"constant","urgency":"low"},"type":"use_support_hours"},"name":"ato-service-staging","scheduled_actions":[null],"support_hours":{"days_of_week":[7,1,2,3,4],"end_time":"19:00:00","start_time":"10:00:00","time_zone":"Asia/Jerusalem","type":"fixed_time_per_day"}}}: timestamp=2022-09-29T13:42:47.272+0300
2022-09-29T13:42:47.283+0300 [INFO]  provider.terraform-provider-pagerduty_v2.6.2: 2022/09/29 13:42:47 [DEBUG] PagerDuty - Preparing PUT request to /services/PAJ8G0E with body: {"service":{"acknowledgement_timeout":null,"alert_creation":"create_alerts_and_incidents","alert_grouping":"intelligent","alert_grouping_parameters":{"type":"intelligent","config":{"timeout":0,"aggregate":""}},"auto_pause_notifications_parameters":{"enabled":true,"timeout":180},"auto_resolve_timeout":null,"description":"Managed by Terraform","escalation_policy":{"id":"P6D1FFL","type":"escalation_policy_reference"},"incident_urgency_rule":{"during_support_hours":{"type":"constant","urgency":"severity_based"},"outside_support_hours":{"type":"constant","urgency":"low"},"type":"use_support_hours"},"name":"customer-center-staging","scheduled_actions":[null],"support_hours":{"days_of_week":[7,1,2,3,4],"end_time":"19:00:00","start_time":"10:00:00","time_zone":"Asia/Jerusalem","type":"fixed_time_per_day"}}}: timestamp=2022-09-29T13:42:47.272+0300
2022-09-29T13:42:47.285+0300 [INFO]  provider.terraform-provider-pagerduty_v2.6.2: 2022/09/29 13:42:47 [DEBUG] PagerDuty API Request Details:
---[ REQUEST ]---------------------------------------
PUT /services/PZW8TJA HTTP/1.1
Host: api.pagerduty.com
User-Agent: heimweh/go-pagerduty(terraform)
Content-Length: 807
Accept: application/vnd.pagerduty+json;version=2
Authorization: Token token=xxxxxxxxxx
Content-Type: application/json
Accept-Encoding: gzip

{
 "service": {
  "acknowledgement_timeout": null,
  "alert_creation": "create_alerts_and_incidents",
  "alert_grouping": "intelligent",
  "alert_grouping_parameters": {
   "type": "intelligent",
   "config": {
    "timeout": 0,
    "aggregate": ""
   }
  },
  "auto_pause_notifications_parameters": {
   "enabled": true,
   "timeout": 180
  },
  "auto_resolve_timeout": null,
  "description": "Managed by Terraform",
  "escalation_policy": {
   "id": "P6D1FFL",
   "type": "escalation_policy_reference"
  },
  "incident_urgency_rule": {
   "during_support_hours": {
    "type": "constant",
    "urgency": "severity_based"
   },
   "outside_support_hours": {
    "type": "constant",
    "urgency": "low"
   },
   "type": "use_support_hours"
  },
  "name": "ato-service-staging",
  "scheduled_actions": [
   null
  ],
  "support_hours": {
   "days_of_week": [
    7,
    1,
    2,
    3,
    4
   ],
   "end_time": "19:00:00",
   "start_time": "10:00:00",
   "time_zone": "Asia/Jerusalem",
   "type": "fixed_time_per_day"
  }
 }
}

**Comment by pdecat** [comment]
Thanks for the logs.

The parameters are clearly sent to the PagerDuty API:
```
"auto_pause_notifications_parameters": {
"enabled": true,
"timeout": 180
},
```

Could you also share the response to the `PUT /services/PZW8TJA` request?

**Comment by pdecat** [comment]
Hi @elihai-risk, this is the *request* to the API, not the *response* from the API.

You should have something just after that in the logs.

**Comment by elihai-risk** [comment]
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Access-Control-Allow-Headers: Authorization, Content-Type, AuthorizationOauth, X-EARLY-ACCESS
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers:
Access-Control-Max-Age: 1728000
Cache-Control: max-age=0, private, must-revalidate
Content-Type: application/json
Date: Thu, 29 Sep 2022 10:41:53 GMT
Etag: W/"307d6d206b82b9b6a923e34a97574eee"
Feature-Policy: accelerometer 'none'; camera 'none'; geolocation 'none'; gyroscope 'none'; magnetometer 'none'; microphone 'none'; payment 'none'; usb 'none'
Referrer-Policy: strict-origin-when-cross-origin
Referrer-Policy: strict-origin-when-cross-origin
Server: nginx
Strict-Transport-Security: max-age=31536000; includeSubDomains
Vary: Accept-Encoding
X-Content-Type-Options: nosniff
X-Request-Id: 890e312363ad1e4559271d5ca9e34ac0
X-Xss-Protection: 1; mode=block

{
 "service": {
  "id": "PZW8TJA",
  "name": "ato-service-staging",
  "description": "Managed by Terraform",
  "created_at": "2022-03-30T16:22:12+03:00",
  "updated_at": "2022-03-30T16:22:12+03:00",
  "status": "active",
  "teams": [
   {
    "id": "PA3L24N",
    "type": "team_reference",
    "summary": "Account Protection",
    "self": "https://api.pagerduty.com/teams/PA3L24N",
    "html_url": "https://riskified.pagerduty.com/teams/PA3L24N"
   }
  ],
  "alert_creation": "create_alerts_and_incidents",
  "addons": [],
  "scheduled_actions": [],
  "support_hours": {
   "type": "fixed_time_per_day",
   "time_zone": "Asia/Jerusalem",
   "days_of_week": [
    7,
    1,
    2,
    3,
    4
   ],
   "start_time": "10:00:00",
   "end_time": "19:00:00"
  },
  "last_incident_timestamp": null,
  "escalation_policy": {
   "id": "P6D1FFL",
   "type": "escalation_policy_reference",
   "summary": "Account Protection - low urgency",
   "self": "https://api.pagerduty.com/escalation_policies/P6D1FFL",
   "html_url": "https://riskified.pagerduty.com/escalation_policies/P6D1FFL"
  },
  "incident_urgency_rule": {
   "type": "use_support_hours",
   "during_support_hours": {
    "type": "constant",
    "urgency": "severity_based"
   },
   "outside_support_hours": {
    "type": "constant",
    "urgency": "low"
   }
  },
  "acknowledgement_timeout": null,
  "auto_resolve_timeout": null,
  "alert_grouping": "intelligent",
  "alert_grouping_timeout": null,
  "alert_grouping_parameters": {
   "type": "intelligent",
   "config": null
  },
  "integrations": [],
  "response_play": null,
  "type": "service",
  "summary": "ato-service-staging",
  "self": "https://api.pagerduty.com/services/PZW8TJA",
  "html_url": "https://riskified.pagerduty.com/service-directory/PZW8TJA"
 }
}

**Comment by pdecat** [comment]
Ok, thanks for the logs.

So in your case, the PagerDuty API is not returning the `auto_pause_notifications_parameters` parameter block.

I just re-ran the acceptance test, and it does work properly:

```
# TF_LOG=debug make testacc TEST=./pagerduty/ TESTARGS='-run=TestAccPagerDutyService_AutoPauseNotificationsParameters'
...
2022/09/29 16:12:01 [DEBUG] PagerDuty API Request Details:
---[ REQUEST ]---------------------------------------
POST /services HTTP/1.1
Host: api.pagerduty.com
User-Agent: heimweh/go-pagerduty(terraform)
Content-Length: 323
Accept: application/vnd.pagerduty+json;version=2
Authorization: Token token=******
Content-Type: application/json
Accept-Encoding: gzip

{
 "service": {
  "acknowledgement_timeout": 1800,
  "alert_creation": "create_alerts_and_incidents",
  "alert_grouping": null,
  "auto_pause_notifications_parameters": {
   "enabled": true,
   "timeout": 300
  },
  "auto_resolve_timeout": 1800,
  "description": "foo",
  "escalation_policy": {
   "id": "PYIKCSU",
   "type": "escalation_policy_reference"
  },
  "name": "tf-tmz1x"
 }
}

-----------------------------------------------------
2022/09/29 16:12:01 [DEBUG] PagerDuty API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 201 Created
Content-Length: 1045
Access-Control-Allow-Headers: Authorization, Content-Type, AuthorizationOauth, X-EARLY-ACCESS
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers:
Access-Control-Max-Age: 1728000
Cache-Control: max-age=0, private, must-revalidate
Content-Type: application/json
Date: Thu, 29 Sep 2022 14:12:01 GMT
Etag: W/"c5cc73a1f5e5667add6272c2485e7f29"
Feature-Policy: accelerometer 'none'; camera 'none'; geolocation 'none'; gyroscope 'none'; magnetometer 'none'; microphone 'none'; payment 'none'; usb 'none'
Referrer-Policy: strict-origin-when-cross-origin
Referrer-Policy: strict-origin-when-cross-origin
Server: nginx
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Content-Type-Options: nosniff
X-Request-Id: bca51a7dd9e03dea3fd801b36d0ec80e
X-Xss-Protection: 1; mode=block

{
 "service": {
  "id": "P8NHZX9",
  "name": "tf-tmz1x",
  "description": "foo",
  "created_at": "2022-09-29T16:12:01+02:00",
  "updated_at": "2022-09-29T16:12:01+02:00",
  "status": "active",
  "teams": [],
  "alert_creation": "create_alerts_and_incidents",
  "addons": [],
  "scheduled_actions": [],
  "support_hours": null,
  "last_incident_timestamp": null,
  "escalation_policy": {
   "id": "PYIKCSU",
   "type": "escalation_policy_reference",
   "summary": "tf-b7xyf",
   "self": "https://api.pagerduty.com/escalation_policies/PYIKCSU",
   "html_url": "https://dev-myorg.pagerduty.com/escalation_policies/PYIKCSU"
  },
  "incident_urgency_rule": {
   "type": "constant",
   "urgency": "high"
  },
  "acknowledgement_timeout": 1800,
  "auto_resolve_timeout": 1800,
  "auto_pause_notifications_parameters": {
   "enabled": true,
   "timeout": 300
  },
  "alert_grouping": null,
  "alert_grouping_timeout": null,
  "alert_grouping_parameters": {
   "type": null,
   "config": null
  },
  "integrations": [],
  "response_play": null,
  "type": "service",
  "summary": "tf-tmz1x",
  "self": "https://api.pagerduty.com/services/P8NHZX9",
  "html_url": "https://dev-myorg.pagerduty.com/service-directory/P8NHZX9"
 }
}
-----------------------------------------------------
```

Notice how the `auto_pause_notifications_parameters` is part of both the request *and* the response. However, that's a POST request so it is a bit different from your case.

Tweaking the acceptance test, I also confirmed updates to existing services (PUT request), like seems your case, do work:

```
---[ REQUEST ]---------------------------------------
PUT /services/PH7ONRC HTTP/1.1
Host: api.pagerduty.com
User-Agent: heimweh/go-pagerduty(terraform)
Content-Length: 384
Accept: application/vnd.pagerduty+json;version=2
Authorization: Token token=******
Content-Type: application/json
Accept-Encoding: gzip

{
 "service": {
  "acknowledgement_timeout": 1800,
  "alert_creation": "create_alerts_and_incidents",
  "alert_grouping": null,
  "auto_pause_notifications_parameters": {
   "enabled": true,
   "timeout": 300
  },
  "auto_resolve_timeout": 1800,
  "description": "foo",
  "escalation_policy": {
   "id": "PRI3HSH",
   "type": "escalation_policy_reference"
  },
  "incident_urgency_rule": {
   "type": "constant",
   "urgency": "high"
  },
  "name": "tf-6i9jw"
 }
}

-----------------------------------------------------
2022/09/29 16:20:26 [DEBUG] PagerDuty API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Access-Control-Allow-Headers: Authorization, Content-Type, AuthorizationOauth, X-EARLY-ACCESS
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers:
Access-Control-Max-Age: 1728000
Cache-Control: max-age=0, private, must-revalidate
Content-Type: application/json
Date: Thu, 29 Sep 2022 14:20:26 GMT
Etag: W/"5fc1b7d303bb4b9b76ba3db103968f61"
Feature-Policy: accelerometer 'none'; camera 'none'; geolocation 'none'; gyroscope 'none'; magnetometer 'none'; microphone 'none'; payment 'none'; usb 'none'
Referrer-Policy: strict-origin-when-cross-origin
Referrer-Policy: strict-origin-when-cross-origin
Server: nginx
Strict-Transport-Security: max-age=31536000; includeSubDomains
Vary: Accept-Encoding
X-Content-Type-Options: nosniff
X-Request-Id: 70383286a32de80308bf97d1228223d1
X-Xss-Protection: 1; mode=block

{
 "service": {
  "id": "PH7ONRC",
  "name": "tf-6i9jw",
  "description": "foo",
  "created_at": "2022-09-29T16:20:18+02:00",
  "updated_at": "2022-09-29T16:20:26+02:00",
  "status": "active",
  "teams": [],
  "alert_creation": "create_alerts_and_incidents",
  "addons": [],
  "scheduled_actions": [],
  "support_hours": null,
  "last_incident_timestamp": null,
  "escalation_policy": {
   "id": "PRI3HSH",
   "type": "escalation_policy_reference",
   "summary": "tf-yzp0v",
   "self": "https://api.pagerduty.com/escalation_policies/PRI3HSH",
   "html_url": "https://dev-myorg.pagerduty.com/escalation_policies/PRI3HSH"
  },
  "incident_urgency_rule": {
   "type": "constant",
   "urgency": "high"
  },
  "acknowledgement_timeout": 1800,
  "auto_resolve_timeout": 1800,
  "auto_pause_notifications_parameters": {
   "enabled": true,
   "timeout": 300
  },
  "alert_grouping": null,
  "alert_grouping_timeout": null,
  "alert_grouping_parameters": {
   "type": null,
   "config": null
  },
  "integrations": [],
  "response_play": null,
  "type": "service",
  "summary": "tf-6i9jw",
  "self": "https://api.pagerduty.com/services/PH7ONRC",
  "html_url": "https://dev-myorg.pagerduty.com/service-directory/PH7ONRC"
 }
}
-----------------------------------------------------
```

Notice too how the `auto_pause_notifications_parameters` is part of both the request *and* the response.

I suggest that you open a support case to PagerDuty pointing them to this issue.

PS: I'm just a contributor here, so I have no idea what's going on the server side of the Pagerduty API.

**Comment by imjaroiswebdev** [comment]
@elihai-risk A patch for solving is now available on version [v2.12.1](https://github.com/PagerDuty/terraform-provider-pagerduty/releases/tag/v2.12.1) of PagerDuty Terraform Provider. Hey @pdecat thank you so much again for your frequent interest on supporting this Terraform Provider users and development lifecycle 🎉 . On top that, I encourage you to do the upgrade and if you still face the issue, please don't hesitate on re-opening this issue ✌🏽 😎
